src/overlays/actors/ovl_En_Niw/z_en_niw.c

```diff
@@ -21,52 +40,1012 @@ const ActorInit En_Niw_InitVars = {
+void EnNiw_Upset(EnNiw* this, GlobalContext* globalCtx) {
+    // assumption: CuccoStorm is split into smaller parts because it used to be a cutscene in OOT
+    this->sfxTimer1 = 100;
+    if (this->unkTimer252 == 0) {
+        this->unkTimer252 = 60;
+        this->unkTimer24C = 10;
+        this->unk29C = 4;
+        this->unknownState28E = 2;
+        this->actionFunc = EnNiw_SetupCuccoStorm;
+    }
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Niw_0x80891060/func_80892274.asm")
+    func_80891320(this, globalCtx, this->unk29C);
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Niw_0x80891060/func_808922D0.asm")
+// the long crow with head back before they descend
+void EnNiw_SetupCuccoStorm(EnNiw* this, GlobalContext* globalCtx) {
+    f32 viewY;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Niw_0x80891060/func_80892390.asm")
+    this->sfxTimer1 = 100;
+    if (this->unkTimer252 == 40) {
+        viewY = 14000.0f;
+        this->unk264[0] = 10000.0f;
+        this->unk264[7] = viewY;
+        if (0) {}
+        this->unk264[5] = viewY;
+        this->unk264[6] = 0.0f;
+        this->unk264[8] = 0.0f;
+        this->unk264[1] = 0.0f;
+        this->unk264[2] = 0.0f;
+        this->unkTimer24C = 10;
+        Audio_PlayActorSound2(&this->actor, NA_SE_EV_CHICKEN_CRY_M); // crow
+    }
+    if (this->unkTimer252 == 0) {
+        this->unkTimer252 = 10;
+        this->yawTowardsPlayer = this->actor.yawTowardsPlayer;
+        this->actor.flags &= ~0x1; // targetable OFF
+        this->unknownState28E = 3;
+        this->actionFunc = EnNiw_CuccoStorm;
+    }
+    func_80891320(this, globalCtx, this->unk29C);
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Niw_0x80891060/func_80892414.asm")
+void EnNiw_CuccoStorm(EnNiw* this, GlobalContext* globalCtx) {
+    EnNiw_SpawnAttackNiw(this, globalCtx);
+    if (this->unkTimer252 == 1) {
+        this->actor.speedXZ = 3.0f;
+        this->unk29A = Rand_ZeroFloat(1.99f);
+        this->unkTimer250 = 0;
+        this->unkTimer24E = this->unkTimer250;
+        this->unkTimer24C = this->unkTimer250;
+    } else {
+        func_808917F8(this, globalCtx, 1);
+    }
+}
 
+void EnNiw_SetupRunning(EnNiw* this) {
+    SkelAnime_ChangeAnim(&this->skelanime, &D_060000E8, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_060000E8.common), 0,
+                         -10.0f);
+    this->unk29A = Rand_ZeroFloat(1.99f);
+    this->unknownState28E = 7;
+    this->actionFunc = func_808924B0; // running away
+    this->actor.speedXZ = 4.0f;
+}
+
+#ifdef NON_MATCHING
+// bad regalloc, center of first if block
+// actionfunc: running away from link
+void func_808924B0(EnNiw* this, GlobalContext* globalCtx) {
+    ActorPlayer* player = PLAYER;
+    Vec3f tempVec3f;
+    s16 temp298;
+    f32 dX;
+    f32 dZ;
+
+    // it actually wants to copy to stack... not modify, then pass to veccopy
+    // and it does it BEFORE the if block, this is just unoptimized.
+    tempVec3f = D_808934E8;
```

Comment:
```suggestion
void func_808924B0(EnNiw* this, GlobalContext* globalCtx) {
    static Vec3f D_808934E8 = { 90000.0f, 90000.0f, 90000.0f, };
    ActorPlayer* player = PLAYER;
    Vec3f tempVec3f;
    s16 temp298;
    f32 dX;
    f32 dZ;

```
I suspect this may be a case where you actually have to bring the data into the function itself. But I could be wrong. Note that this means you need to have the `static s32 pad = 0;` anywhere below this function.

Reply:
![moving static into function](https://user-images.githubusercontent.com/42048411/123523482-b1107f00-d678-11eb-898d-e601c495ca4d.png)
This function wasn't matching to begin with, but this looks further from what it was trying to do

Reply:
Oh interesting. And does that include not assigning a temp to the data once it's brought into the function? It's fine to leave it as is otherwise